controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1964,26 +1940,32 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
                         return getOrderedCommittingTxnInLowestEpoch(limit, context)
                                 .thenCompose(list -> {
                                     if (list.isEmpty()) {
-                                        return CompletableFuture.completedFuture(versioned);
+                                        List<VersionedTransactionData> emptyTransactionData = new LinkedList<>();
+                                        return CompletableFuture.completedFuture(new SimpleEntry<>(versioned, emptyTransactionData));
                                     } else {
-                                        Map.Entry<UUID, ActiveTxnRecord> firstEntry = list.get(0);
-                                        ImmutableList.Builder<UUID> txIdList = ImmutableList.builder(); 
-                                        list.forEach(x -> txIdList.add(x.getKey()));
-                                        List<Long> positions = list.stream().map(x -> x.getValue().getCommitOrder())
+                                        ImmutableList.Builder<UUID> txIdList = ImmutableList.builder();
+                                        list.forEach(x -> {
```

Comment:
Isn't there an API on ImmutableList to either wrap or make a copy of `list`? 

Actually where do you use this anyway?

Reply:
we could have done list.stream.map.collect and then made an immutable copy. that would have used 2x the memory.